ui/src/js/storyTelling-page-storyEdit.js

```diff
@@ -207,6 +230,40 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    /* Removes all empty blocks from a given array of story blocks
+     * - "blocks": an array of story blocks
+     * - "requiredValuesLookup": a collection of arrays by block type which
+     *      outlines the values that, if at least one is truthy, will mean a
+     *      particular block is not empty
+     */
+    sjrk.storyTelling.page.storyEdit.removeEmptyBlocks = function (blocks, requiredValuesLookup) {
+        var nonEmptyBlocks = fluid.remove_if(blocks, function (block) {
+            return sjrk.storyTelling.page.storyEdit.isEmptyBlock(block, requiredValuesLookup[block.blockType]);
+        });
+
+        return nonEmptyBlocks;
+    };
+
+    /* Returns true if a block is determined to be empty. Which values determine
+     * whether a block is emtpy depends on the particular block type, but if
+     * they're all falsy then the block is considered empty. If at least one of
+     * those values is truthy, the block is not empty.
+     * - "block": a story block
+     * - "requiredValues": an array of the "required" values of a block, according
+     *                     to the definition above
+     */
+    sjrk.storyTelling.page.storyEdit.isEmptyBlock = function (block, requiredValues) {
+        var isEmptyBlock = true; // assume the block is empty
+
+        fluid.each(requiredValues, function (requiredValue) {
+            if (block[requiredValue]) {
```

Comment:
The `fluid.each` iteration can be stopped once the first non-empty value is hit. What do you think to change line 260 to `return false`?

Reply:
If i change line 260 to `return false`, I'm only returning the callback to fluid.each, aren't I? I think I had tried this when I first wrote these lines

Reply:
Good point and it's true. We won't be able to stop the iteration.

Reply:
Probably try fluid.find() to stop at the first non-empty value.

Reply:
I've made this change using fluid.find_if